### PR TITLE
Expose OIDC config parameters

### PR DESCRIPTION
**Pull request by arvchristos**
This MR exposes OIDC configuration parameters in the default config entrypoint script and the env variables. Essentially, `misp-docker` can now easily work with external SSO providers, a requirement for our use-case.

Settings are configured according to https://github.com/MISP/MISP/blob/2.4/app/Plugin/OidcAuth/README.md

By default, OIDC is disabled but can be enabled with the env variable `OIDC_ENABLE`.

### CODE REVIEW

**Comment by ostefano** [comment]
Thank you @arvchristos , I like this.

We are using OIDC as well, so might take this as an opportunity to merge some additonal logic.
I will iterate later today.

File: core/files/configure_misp.sh [hunk 1]
```diff
@@ -93,6 +103,33 @@ GPGEOF
     sudo -u www-data /var/www/MISP/app/Console/cake Admin setSetting -q "GnuPG.binary" "$(which gpg)"
 }
 
+set_up_oidc() {
+    if [[ "$OIDC_ENABLE" != "true" ]]; then
+        echo "... OIDC authentication disabled"
+        return
+    fi
+
+    sudo -u www-data php /var/www/MISP/tests/modify_config.php modify "{
```
Comment (ostefano) [inline core/files/configure_misp.sh:107]: Note that this is not idempotent (can't recall the exact behavior :S).

Best fix, is to reset the value before setting it.

```
sudo -u www-data php /var/www/MISP/tests/modify_config.php modify "{
            \"Security\": {
                \"auth\": \"\"
            }
}" > /dev/null 
```

File: template.env [hunk 2]
```diff
@@ -87,3 +87,12 @@ SYNCSERVERS_1_KEY=
 
 # Disable IPv6 completely (this setting will persist until the container is removed)
 # DISABLE_IPV6=true
+
+# Enable OIDC authentication, according to https://github.com/MISP/MISP/blob/2.4/app/Plugin/OidcAuth/README.md
+# OIDC_ENABLE=true
+# OIDC_PROVIDER_URL=
+# OIDC_CLIENT_ID=
+# OIDC_CLIENT_SECRET=
+# OIDC_ROLES_PROPERTY=
```
Comment (ostefano) [inline template.env]: I would add some example re the data type

File: template.env [hunk 3]
```diff
@@ -87,3 +87,12 @@ SYNCSERVERS_1_KEY=
...
+# OIDC_ROLES_MAPPING=
```
Comment (ostefano) [inline template.env]: Same thing here.

File: core/files/configure_misp.sh [hunk 4]
```diff
@@ -6,10 +6,20 @@ source /rest_client.sh
 [ -z "$GPG_PASSPHRASE" ] && GPG_PASSPHRASE="passphrase"
 [ -z "$REDIS_FQDN" ] && REDIS_FQDN="redis"
 [ -z "$MISP_MODULES_FQDN" ] && MISP_MODULES_FQDN="http://misp-modules"
+[ -z "$OIDC_PROVIDER_URL" ] && OIDC_PROVIDER_URL="test_provider"
```
Comment (ostefano) [inline core/files/configure_misp.sh]: Not sure whether we want default values if not working

File: core/files/configure_misp.sh (hunk 1 above)
Comment (arvchristos) [reply core/files/configure_misp.sh:107]: I thought of this but this step is already happening at https://github.com/MISP/misp-docker/blob/526c47a6e8b92e1c761030b7a93a95c7b6c85bf3/core/files/configure_misp.sh#L192  which is by default executed and cannot be disabled with a flag.

File: core/files/configure_misp.sh (hunk 4 above)
Comment (ostefano) [reply core/files/configure_misp.sh]: Let me rephrase it. I would rather remove default values, and add some checks when setting up oidc (e.g., if not all values are set, error out).

Comment (arvchristos) [reply core/files/configure_misp.sh]: Done and added a generic  utility which can be used for other configuration functions in the future

Comment (ostefano) [reply core/files/configure_misp.sh]: Nice. I see the output is redirected to stderr. Should we leave it like that?

Comment (arvchristos) [reply core/files/configure_misp.sh]: Actually no, you are totally right, fixed

File: template.env (hunk 2 above)
Comment (arvchristos) [reply template.env]: Fixed

File: template.env (hunk 3 above)
Comment (arvchristos) [reply template.env]: Fixed

**Comment by ostefano** [comment]
LGTM
